tests/Controller/UsersControllerTest.php

```diff
@@ -295,7 +316,38 @@ public function testSetTrustedWithNonAdminIdThrowsException()
         $usersController->setTrusted($request, $db);
     }
 
+    /**
+     * Ensures that if the setAdmin method is called and user_id is a,
+     * non-admin, an exception is thrown
+     *
+     * @return void
+     *
+     * @test
+     */
+    public function testSetAdminWithNonAdminIdThrowsException()
+    {
+        $request = new Request([], ['REQUEST_URI' => "http://api.dev.joind.in/v2.1/users/4/admin", 'REQUEST_METHOD' => 'POST']);
+        $request->user_id = 2;
+        $usersController = new UsersController();
+        $db = $this->getMockBuilder(\PDO::class)->disableOriginalConstructor()->getMock();
+
+        $userMapper = $this->getMockBuilder(UserMapper::class)
+            ->setConstructorArgs([$db,$request])
+            ->getMock();
+
+        $userMapper
+            ->expects($this->once())
+            ->method('isSiteAdmin')
+            ->will($this->returnValue(false));
+
+        $usersController->setUserMapper($userMapper);
 
+        self::expectException(\Exception::class);
```

Comment:
Every Exception is an `Exception`. It may be beneficial to use more specific error classes